File: buzzvil-sdk-v5-ios/buzzvil-sdk-ios-objc/MissionPack/MissionPackViewController.m
@import BuzzvilSDK;
#import "MissionPackViewController.h"

@interface MissionPackViewController()

@property (nonatomic, strong, readonly) BuzzAdBenefitMissionPack * missionPack;

@end

@implementation MissionPackViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  _missionPack = [[BuzzAdBenefitMissionPack alloc] initWithUnitId:@"YOUR_MISSION_PACK_UNIT_ID"];
}

- (void)checkCanOpenMissionPack {
  [_missionPack canOpenOnSuccess:^{
    // missionPack 이 열릴 수 있을 때 호출 됩니다.
  } onFailure:^(NSError * _Nonnull error) {
    // missionPack 이 열릴 수 없을 때 호출 됩니다.
  }];
}

- (void)showMissionPack {
  [_missionPack showOn:self onFailure:^(NSError * _Nonnull error) {
    // missionPack 이 열릴 수 없을 때 호출 됩니다.
  }];
}

@end

File: buzzvil-sdk-v5-ios/buzzvil-sdk-ios-swift/MissionPack/MissionPackViewController.swift
import UIKit
import BuzzvilSDK

class MissionPackViewController: UIViewController {
  let missionPack = BuzzAdBenefitMissionPack(unitId: "YOUR_MISSION_PACK_UNIT_ID")
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    showMissionPack()
  }
  
  // missionPack 사용가능한지 체크하기
  func checkCanOpenMissionPack() {
    missionPack.canOpen {
      print ("MissionPack Can open !")
    } onFailure: { error in
      print ("MissionPack Can't open error: \(error.localizedDescription)")
    }
  }
  
  // missionPack 화면에 표시하기
  func showMissionPack() {
    missionPack.show(on: self, onFailure: { error in
      print ("MissionPack Can't open error: \(error.localizedDescription)")
    })
  }
}

